### DISCUSSION TITLE
Customize shopware-6-client to add new API call

### DISCUSSION THREAD

**Discussion by sourin00** [vuestorefront/shopware-pwa#1804] [Q&A] [answered]
### Customize shopware-6-client to add new API call

_Hi, 
I'm pretty new to Vue & Nuxt.. so the question maybe very basic 😅_ 

**How can I customize/override/extend a service from the shopware-6-client(API client) to have my own implementation which I can use in a composable ?**

**Comment by niklaswolf** [comment, marked as answer]
Hey, 
if I understand your question correctly, you want to do a new API call from a new composable you're creating. For this, you actually don't need to extend the shopware-6-client, but rather use its functionality.

```js
import { getApplicationContext} from "@shopware-pwa/composables"
...
const { apiInstance } = getApplicationContext({
    contextName: "YourOwnContextName"
})
...
async function makeApiCall (){
    const response = await apiInstance.invoke.post(API_ENDPOINT, params)
}
```

`apiInstance` will be a instance from shopware-6-client, so you don't need to worry about authentication and the session-context and so on. 
`apiInstance.invoke.post` will be essentially the post-function from axios, so you can do with it whatever you can do with axios.

I hope this helped :)

**Reply by sourin00** [reply]
Yeah got it! Thanks for the answer 🙂
